bot/src/trader.rs

```diff
@@ -46,7 +44,16 @@ impl Trader {
             Channel::Listed | Channel::PriceChanged => {
                 if let (Some(mean), Some(price)) = (stats.mean_price, data.price) {
                     if Self::is_mean_reliable(&stats) && (price as f64) < BUY_THRESHOLD * mean {
-                        self.handle_purchase(&data, mean).await?;
+                        let list = self.http.fetch_market_data(data.skin_id, 0).await?;
+                        let mut id_lowest = data.id;
```

Comment:
Instead of moving the `String` here, we could either borrow the entire string `&String` or we can get an immutable reference (view) to the string (`&str`) by calling `data.id.as_str()`. Usually, if you just want to immutably reference some `String` (borrow the string without being able to change it), then you use a `&str` primitive type. Strings are a little weird. Think of it as having a `Vec<T>` but you can can get a `slice` (view) of it as `&[T]`